Consul/Configuration.cs

```diff
@@ -0,0 +1,1736 @@
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int? Port { get; set; }
+    }
+
+    public interface IConfigurationEntry
+    {
+        string Kind { get; set; }
+    }
+
+    public class Configuration : IConfigurationEndpoint
+    {
+        private readonly ConsulClient _client;
+
+        internal Configuration(ConsulClient c)
+        {
+            _client = c;
+        }
+
+        /// <summary>
+        ///  This creates or updates the given config entry.
+        /// </summary>
+        /// <param name="q">Write Options</param>
+        /// <param name="configurationEntry">The configuration entry</param>
+        /// <returns>An empty write result</returns>
+        public Task<WriteResult> ApplyConfig<TConfig>(WriteOptions q, TConfig configurationEntry, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Put("/v1/config", configurationEntry, q);
+            return req.Execute(ct);
+        }
+
+        /// <summary>
+        ///  This creates or updates the given config entry.
+        /// </summary>
+        /// <param name="configurationEntry">The configuration entry</param>
+        /// <returns>An empty write result</returns>
+        public Task<WriteResult> ApplyConfig<TConfig>(TConfig configurationEntry, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Put("/v1/config", configurationEntry, WriteOptions.Default);
+            return req.Execute(ct);
+        }
+
+        /// <summary>
+        /// This Retrieves the given config entry.
+        /// </summary>
+        /// <typeparam name="TConfig"></typeparam>
+        /// <param name="kind">The kind of config entry</param>
+        /// <param name="name">The name of config entry</param>
+        /// <param name="q">Query Options</param>
+        /// <param name="ct">Cancellation Token</param>
+        /// <returns>A config entry</returns>
+        public Task<QueryResult<TConfig>> GetConfig<TConfig>(string kind, string name, QueryOptions q, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Get<TConfig>($"/v1/config/{kind}/{name}", q);
+            return req.Execute(ct);
+        }
+
+        /// <summary>
+        /// This Retrieves the given config entry.
+        /// </summary>
+        /// <typeparam name="TConfig"></typeparam>
+        /// <param name="kind">The kind of config entry</param>
+        /// <param name="name">The name of config entry</param>
+        /// <param name="ct">Cancellation Token</param>
+        /// <returns>A config entry</returns>
+        public Task<QueryResult<TConfig>> GetConfig<TConfig>(string kind, string name, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Get<TConfig>($"/v1/config/{kind}/{name}", QueryOptions.Default);
```

Comment:
You can call the other variant of this method from here, so we avoid repeating the code.